src/ui/components/CommandPalette/CommandPalette.tsx

```diff
@@ -87,9 +92,10 @@ const DEFAULT_COMMANDS: readonly CommandKey[] = [
 const COMMAND_HEIGHT = 36;
 const ITEMS_TO_SHOW = 7;
 
-function getShownCommands(searchString: string) {
+function getShownCommands(searchString: string, userInfo: UserInfo) {
+  const availableCommands = COMMANDS.filter(command => !command.internalOnly || userInfo.internal);
   const commands: readonly Command[] = searchString
-    ? filter(COMMANDS as Command[], searchString, { key: "label" })
+    ? filter(availableCommands as Command[], searchString, { key: "label" })
     : COMMANDS;
```

Comment:
The filtered available commands will be completely bypassed if there's an empty search string because of this.

```suggestion
    ? filter(availableCommands as Command[], searchString, { key: "label" })
    : availableCommands;
```